src/app/passenger-list/passenger-list.component.ts

```diff
@@ -1,16 +1,32 @@
 import { Component } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
 
-import { PASSENGERS } from '../../assets/mock-data/mock-passengers';
+import { Passenger } from "src/app/interfaces/passenger"
 
 @Component({
     selector: 'app-passenger-list',
     templateUrl: './passenger-list.component.html',
     styleUrls: ['./passenger-list.component.scss']
 })
 export class PassengerListComponent {
-    constructor() { }
 
-    ngOnInit() { }
+    private passengersUrl = "/assets/mock-data/mockPassengers.json";
 
-    passengers = PASSENGERS;
+    isDataAvailable = false;
+
+    passengers: any;
+
+    constructor(private http: HttpClient) { }
+
+    ngOnInit() {
+        this.getJSON(this.passengersUrl).subscribe((data) => {
+            this.passengers = <Passenger[]>data.passengers;
+            this.isDataAvailable = true;
+        })
+    }
+
+    private getJSON(url: string): Observable<any> {
+        return this.http.get(url);
+    }
```

Comment:
This is better than before, but still, not the way it should be. In Angular applications you would have services for retrieving data, instead of calling HttpClient directly from the components